microsoft/testsuites/core/storage.py

```diff
@@ -345,6 +346,46 @@ def hot_add_disk_parallel_premium_ssd(self, log: Logger, node: Node) -> None:
             log, node, DiskType.PremiumSSDLRS, self.DEFAULT_DISK_SIZE_IN_GB
         )
 
+    @TestCaseMetadata(
+        description="""
+        This test case will verify mount azure nfs on guest successfully.
+        """,
+        timeout=TIME_OUT,
+        requirement=simple_requirement(supported_features=[Nfs]),
+        priority=2,
+    )
+    def verify_azure_file_share_nfs(self, log: Logger, node: Node) -> None:
+        nfs = node.features[Nfs]
+        mount_dir = "/mount/azure_share"
+        try:
+            nfs.create_share()
+            storage_account_name = nfs.storage_account_name
+            mount_nfs = f"{storage_account_name}.file.core.windows.net"
+            server_shared_dir = f"{nfs.storage_account_name}/{nfs.file_share_name}"
+            mount_success = False
+            # retry if private connection doesn't take effect immediately
+            timer = create_timer()
+            while timer.elapsed(False) < 60:
+                try:
+                    node.tools[NFSClient].setup(
+                        mount_nfs,
+                        server_shared_dir,
+                        mount_dir,
+                    )
+                    mount_success = True
+                    break
+                except Exception as identifier:
+                    time.sleep(5)
+                    log.debug(
+                        f"{identifier.__class__.__name__}: {identifier}. Retry..."
```

Comment:
1. It looks like happening in the `setup` method. So the retry should be in the setup method. Another way is to add retry decorator on the `setup` method. It can save this whole logic.
2. The message doesn't specify where it's from. So it needs to see like waiting for something effect.

Reply:
move retry into mount, and update the exception message